### PR TITLE
Allow to run a specific rustdoc-js* test

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Mark-Simulacrum

### CODE REVIEW

File: src/bootstrap/test.rs [hunk 1]
```diff
@@ -783,6 +783,13 @@ impl Step for RustdocJSStd {
                 .arg(builder.doc_out(self.target))
                 .arg("--test-folder")
                 .arg(builder.src.join("src/test/rustdoc-js-std"));
+            for path in &builder.paths {
+                if let Some(name) = path.file_name().and_then(|f| f.to_str()) {
+                    if name.ends_with(".js") {
```
Comment (Mark-Simulacrum) [inline src/bootstrap/test.rs:788]: This is probably not quite right -- the typical filter can be found in this file, if you search for "Get test-args by striping suite path" -- the logic is somewhat more complicated. Extracting that into a helper function seems like a good idea to me.

Comment (GuillaumeGomez) [reply src/bootstrap/test.rs:788]: Indeed! I'll extract the function then.

Comment (GuillaumeGomez) [reply src/bootstrap/test.rs:788]: I can't re-use the same code because I need the full path. However, I think I 
can do something similar.

Comment (GuillaumeGomez) [reply src/bootstrap/test.rs:788]: I can't re-use the same code because I need the full path. However, I think I can do something similar.

**Comment by GuillaumeGomez** [comment]
@Mark-Simulacrum Thanks for the suggestion! Thanks to it, I was able to clean another test-suite argument as well.

File: src/bootstrap/test.rs [hunk 2]
```diff
@@ -938,8 +947,11 @@ impl Step for RustdocGUI {
             .arg("--tests-folder")
             .arg(builder.build.src.join("src/test/rustdoc-gui"));
         for path in &builder.paths {
-            if let Some(name) = path.file_name().and_then(|f| f.to_str()) {
-                if name.ends_with(".goml") {
+            if let Some(p) = util::is_valid_test_suite_arg(path, "src/test/rustdoc-gui", builder) {
+                if !p.ends_with(".goml") {
+                    continue;
```
Comment (Mark-Simulacrum) [inline src/bootstrap/test.rs:952]: Why this filter?

I think we should give an error message if someone passes a non-goml path inside src/test/rustdoc-gui. (And similar for the other cases where you added a similar filter).

Comment (GuillaumeGomez) [reply src/bootstrap/test.rs:952]: Fine by me!

**Comment by GuillaumeGomez** [comment]
Added a panic in case the file extension wasn't the expected one.

**Comment by Mark-Simulacrum** [comment]
r=me with commits squashed

Thanks!

**Pull request merged**